apps/www/_blog/2022-08-15-launch-week-five-cli-v1-and-admin-api-beta.mdx

```diff
@@ -0,0 +1,130 @@
+
+In the long-term we hope to consolidate on a single diff-ing tool which is perfect, but diff-ing is hard so we would need your help and feedback to improve tooling.
+
+### Manual migrations
+
+For manual migrations, you can now test them out locally without the fear of losing unsaved data. Run the following commands to clone your local database to a new branch.
+
+```bash
+supabase db branch new my_branch
+supabase db branch switch my_branch
+```
+
+Now you can run any DDL statements from Studio UI’s SQL Editor. To undo the changes, simply switch back and delete the new branch.
+
+```bash
+supabase db branch switch main
+supabase db branch delete my_branch
+```
+
+## CI / CD
+
+Automating migrations and tests on your CI / CD pipeline gives developers more confidence that each PR contains a working migration script. CLI v1 focuses on both local and [GitHub Action](https://github.com/supabase/setup-cli) support for the following workflows.
+
+Testing all migrations on a fresh local database
+
+```bash
+supabase init
+supabase start
+```
+
+Releasing schema changes to staging and production
+
+```bash
+$ supabase link --project-ref <project-id>
+$ supabase db push
+```
+
+We created an [example project](https://github.com/supabase/supabase-action-example) showing how to set up GitHub Actions to test and migrate with Supabase CLI v1.
+
+## Type generation
+
+You can now generate TypeScript types using the CLI:
+
+```bash
+# in a project set up with the CLI:
+supabase gen types typescript --local
+supabase gen types typescript --db-url $SUPABASE_DB_URL
+```
+
+Of course, the types aren’t very useful on its own, you need some way to consume it. For that, stay tuned for tomorrow! 😉
+
+# Admin API
+
+The CLI is the first consumer of our new admin API. Over the next few weeks, we’ll be adding all the endpoints needed to programmatically add, remove, and update your Supabase projects. This is ideal for CI/CD workflows and spinning up test environments.
+
+Here’s an example of how you can deploy a new project from the command line, you can generate your access tokens from [Supabase dashboard](https://app.supabase.com/account/tokens).
+
+```bash
+curl 'https://api.supabase.com/v1/projects' \
+  -H 'Authorization: Bearer <[your-access-token](https://app.supabase.com/account/tokens)>' \
+  -H "Content-Type: application/json" \
+  -d '{"name": "my-project", "organization_id": "cool-green-pqdr0qc", "region": "us-east-1", "plan": "free", "db_pass": "********"}'
+
+# Response JSON:
+{
+  "id": "abcdefghijklmnopqrst",
+  "organization_id": "cool-green-pqdr0qc",
+  "name": "hello",
+  "region": "us-east-1",
+  "created_at": "2022-08-12T17:37:11.88819Z"
+}
```

Comment:
Can you move the response JSON to its own code block? The copy button for the curl snippet also copies the JSON since it's in the same code block.

<img width="555" alt="Screen Shot 2022-08-14 at 10 07 46 PM" src="https://user-images.githubusercontent.com/7026076/184579695-fd93f2f2-4b74-4d98-84d3-b74404c3731d.png">

Reply:
good idea